pkg/controller/configmap/configmap_controller.go

```diff
@@ -0,0 +1,129 @@
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	"sigs.k8s.io/controller-runtime/pkg/source"
+)
+
+const (
+	controllerName   = "configmap"
+	configMapName    = "cco-trusted-ca"
+	configMapKeyName = "ca-bundle.crt"
+	ccoNamespace     = "openshift-cloud-credential-operator"
+)
+
+// Add creates a new ConfigMap controller and adds it to the manager.
+func Add(mgr manager.Manager) error {
+	return AddToManager(mgr, NewReconciler(mgr))
+}
+
+// NewReconciler returns a new reconcile.Reconciler
+func NewReconciler(mgr manager.Manager) *ReconcileConfigMap {
+	return &ReconcileConfigMap{
+		Client: mgr.GetClient(),
+		logger: log.WithField("controller", controllerName),
+		exit:   exitFunc,
+	}
+}
+
+// AddToManager adds a new Controller to mgr with r as the reconcile.Reconciler
+//func AddToManager(mgr manager.Manager, r *ReconcileConfigMap) error {
+func AddToManager(mgr manager.Manager, r reconcile.Reconciler) error {
+	c, err := controller.New(controllerName, mgr, controller.Options{Reconciler: r})
+	if err != nil {
+		log.WithField("controller", controllerName).WithError(err).Error("Error creating controller")
+		return err
+	}
+
+	// Watch for changes to the configMap containing the trusted CAs
+	p := predicate.Funcs{
+		UpdateFunc: func(e event.UpdateEvent) bool {
+			return configMapCheck(e.MetaNew)
+		},
+		CreateFunc: func(e event.CreateEvent) bool {
+			return configMapCheck(e.Meta)
+		},
+		DeleteFunc: func(e event.DeleteEvent) bool {
+			return configMapCheck(e.Meta)
+		},
+	}
+	err = c.Watch(&source.Kind{Type: &corev1.ConfigMap{}}, &handler.EnqueueRequestForObject{}, p)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func configMapCheck(cm metav1.Object) bool {
+	return cm.GetName() == configMapName && cm.GetNamespace() == ccoNamespace
+}
+
+var _ reconcile.Reconciler = &ReconcileConfigMap{}
+
+// ReconcileConfigMap reconciles a ConfigMap against the on-disk contents.
+type ReconcileConfigMap struct {
+	client.Client
+	logger            log.FieldLogger
+	configMapDataHash string
+	// Allow testing to catch an exit call from the controller
+	exit func()
+}
+
+// Reconcile checks for changes to the contents of the certificate authority configMap
+func (r *ReconcileConfigMap) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	cm := &corev1.ConfigMap{}
+	if err := r.Get(context.TODO(), request.NamespacedName, cm); err != nil {
+		r.logger.WithError(err).Error("failed to read in configmap")
+		return reconcile.Result{}, err
+	}
+
+	cmHash := fmt.Sprintf("%x", md5.Sum([]byte(cm.Data[configMapKeyName])))
+
+	if r.configMapDataHash == "" {
+		r.logger.Info("Saving hash of configmap containing CA data")
```

Comment:
When initializing r.logger lets add the configmap namespace/name, and the key as context fields.

Maybe "saving hash of proxy CA configmap".